hibernate-reactive-core/src/main/java/org/hibernate/reactive/loader/ReactiveLoaderBasedResultSetProcessor.java

```diff
@@ -0,0 +1,187 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.loader;
+
+import org.hibernate.HibernateException;
+import org.hibernate.dialect.pagination.LimitHelper;
+import org.hibernate.engine.internal.TwoPhaseLoad;
+import org.hibernate.engine.spi.*;
+import org.hibernate.event.service.spi.EventListenerRegistry;
+import org.hibernate.event.spi.*;
+import org.hibernate.internal.CoreLogging;
+import org.hibernate.internal.CoreMessageLogger;
+import org.hibernate.loader.plan.exec.query.spi.NamedParameterContext;
+import org.hibernate.loader.spi.AfterLoadAction;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.persister.entity.Loadable;
+import org.hibernate.reactive.ReactiveLoaderBasedLoader;
+import org.hibernate.reactive.util.impl.CompletionStages;
+import org.hibernate.transform.ResultTransformer;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * A {@link ReactiveResultSetProcessor} intended to be used by {@link ReactiveLoaderBasedLoader} implementations.
```

Comment:
Does the logic in this class come from some specific place in Hibernate core?

Reply:
I've added Javadoc indicating the methods upon which they are based.

Reply:
Great, thanks